src/Operation/Recommendation/BatchedCategoryMerchandising.php

```diff
@@ -0,0 +1,157 @@
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2020 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Operation\Recommendation;
+
+use Nosto\NostoException;
+use Nosto\Request\Http\Exception\AbstractHttpException;
+use Nosto\Request\Http\Exception\HttpResponseException;
+use Nosto\Result\Graphql\Recommendation\CategoryMerchandisingResult;
+use Nosto\Types\Signup\AccountInterface;
+use Nosto\Util\Recommendation;
+
+class BatchedCategoryMerchandising
+{
+    const HARD_LIMIT = 250;
+
+    /** @var string $category */
+    private $category;
+
+    /** @var IncludeFilters */
+    private $includeFilters;
+
+    /** @var ExcludeFilters */
+    private $excludeFilters;
+
+    /** @var int */
+    private $skipPages;
+
+    /** @var AccountInterface */
+    private $account;
+
+    /** @var string */
+    private $customerId;
+
+    /** @var string */
+    private $activeDomain;
+
+    /** @var string */
+    private $customerBy;
+
+    /** @var bool */
+    private $previewMode;
+
+    /** @var int */
+    private $limit;
+
+    /**
+     * CategoryMerchandising constructor.
+     * @param AccountInterface $account
+     * @param $customerId
+     * @param $category
+     * @param $skipPages
+     * @param IncludeFilters $includeFilters
+     * @param ExcludeFilters $excludeFilters
+     * @param string $activeDomain
+     * @param string $customerBy
+     * @param bool $previewMode
+     * @param int $limit
+     */
+    public function __construct(
+        AccountInterface $account,
+        $customerId,
+        $category,
+        $skipPages,
+        IncludeFilters $includeFilters,
+        ExcludeFilters $excludeFilters,
+        $activeDomain = '',
+        $customerBy = CategoryMerchandising::IDENTIFIER_BY_CID,
+        $previewMode = false,
+        $limit = CategoryMerchandising::DEFAULT_LIMIT
+    ) {
+        $this->category = $category;
+        $this->skipPages = $skipPages;
+        $this->includeFilters = $includeFilters;
+        $this->excludeFilters = $excludeFilters;
+        $this->account = $account;
+        $this->customerId = $customerId;
+        $this->activeDomain = $activeDomain;
+        $this->customerBy = $customerBy;
+        $this->previewMode = $previewMode;
+        $this->limit = $limit;
+    }
+
+    /**
+     * Splits category merchandising calls based on hard limit (250) and given limit
+     *
+     * @return CategoryMerchandisingResult
+     * @throws AbstractHttpException
+     * @throws HttpResponseException
+     * @throws NostoException
+     */
+    public function execute()
+    {
+        $batchCount = 1;
+        $limit = $this->limit;
+        if ($this->limit >= self::HARD_LIMIT) {
```

Comment:
There might be a case when the limit is 0. (https://github.com/Nosto/nosto-magento2-cmp/blob/develop/Model/Service/Recommendation/StateAwareCategoryService.php#L264)

Also there another case  to keep in mind. One ore more products are blacklisted on Nosto side.  It might result in one or more pages missing in Nosto, but probably it wont affect the end result.